pinot-core/src/main/java/org/apache/pinot/core/periodictask/BasePeriodicTask.java

```diff
@@ -87,13 +94,16 @@ public final synchronized void start() {
       LOGGER.warn("Task: {} is already started", _taskName);
       return;
     }
-    _started = true;
 
     try {
       setUpTask();
     } catch (Exception e) {
       LOGGER.error("Caught exception while setting up task: {}", _taskName, e);
     }
+
+    // mark _started as true only after state has completely initialized, so that run method doesn't end up seeing
```

Comment:
Hmm.. The run() method should never be called unless the start() method has completely initialized, right? Or, are you trying to handle the case where we are trying to start the task after a stop() has been called? In case of latter, please keep a variable called  `_stopped`

Reply:
This class is used by multiple threads, so it's a good idea to make it thread safe irrespective of how it is used. The API shouldn't allow a situation where run method ends up being called on partially initialized state.

Reply:
ok